doc/src/performance_tuning.rst

```diff
@@ -0,0 +1,151 @@
+.. highlight:: none
+
+******************
+Performance Tuning
+******************
+
+OpenMP Backend
+==============
+
+libxsmm
+-------
+
+Although PyFR will usually function without libxsmm it is nevertheless
+*highly* recommended and can have a very large impact on performance.
+
+AVX-512
+-------
+
+When running on an AVX-512 capable CPU Clang and GCC will, by default,
+only make use of 256-bit vectors.  Given that the kernels in PyFR
+benefit meaningfully from longer vectors it is desirable to override
+this behaviour.  This can be accomplished through the ``cflags`` key
+as::
+
+        [backend-openmp]
+        cflags = -mprefer-vector-width=512
+
+Cores vs. threads
+-----------------
+
+PyFR does not typically derive any benefit from SMT.  As such the
+number of OpenMP threads should be chosen to be equal to the number of
+physical cores.
+
+MPI processes vs. OpenMP threads
```

Comment:
Nowhere in the docs do we say that the environment variable OMP_NUM_THREADS can be used to set the threads in pyfr. This might be useful information to either give here or somewhere else.

Reply:
It is a bit more subtle, as OMP_NUM_THREADS is not always the best way to go about it depending on the OpenMP runtime/MPI library.